sfmutils/exporter.py

```diff
@@ -100,21 +102,25 @@ def on_message(self):
                 }
                 # Other possibilities: XML, databases, HDFS
                 if export_format == "json_full":
-                    filepath = "{}.json".format(base_filepath)
-                    log.info("Exporting to %s", filepath)
-                    self._full_json_export(warc_paths, filepath, dedupe, item_date_start, item_date_end, seed_uids)
+                    self._full_json_export(warc_paths, base_filepath, dedupe, item_date_start, item_date_end, seed_uids,
+                                           export_segment_size)
                 elif export_format == "dehydrate":
-                    table = self.table_cls(warc_paths, dedupe, item_date_start, item_date_end, seed_uids)
-                    filepath = "{}.txt".format(base_filepath)
-                    log.info("Exporting to %s", filepath)
-                    petl.totext(table, filepath, template="{{{}}}\n".format(table.id_field()))
+                    tables = self.table_cls(warc_paths, dedupe, item_date_start, item_date_end, seed_uids)
+                    tables.set_segment_size(export_segment_size)
+                    for idx, table in enumerate(tables):
+                        filepath = "{}_{}.txt".format(base_filepath, idx + 1)
+                        log.info("Exporting to %s", filepath)
+                        petl.totext(table, filepath, template="{{{}}}\n".format(tables.id_field()))
                 elif export_format in export_formats:
-                    table = self.table_cls(warc_paths, dedupe, item_date_start, item_date_end, seed_uids)
-                    filepath = "{}.{}".format(base_filepath, export_formats[export_format][0])
-                    log.info("Exporting to %s", filepath)
-                    export_formats[export_format][1](table, filepath)
-                    if export_format == 'html':
-                        self._file_fix(filepath, prefix="<html><head><meta charset='utf-8'></head>\n", suffix="</html>")
+                    tables = self.table_cls(warc_paths, dedupe, item_date_start, item_date_end, seed_uids)
+                    tables.set_segment_size(export_segment_size)
+                    for idx, table in enumerate(tables):
+                        filepath = "{}_{}.{}".format(base_filepath, idx + 1, export_formats[export_format][0])
```

Comment:
For the filename use a dash followed by a two digit padded number, for example my_export-01.txt.